flexmeasures/cli/data_delete.py

```diff
@@ -196,4 +203,41 @@ def delete_prognoses(
     depopulate_prognoses(app.db, sensor_id)
 
 
+@fm_dev_delete_data.command("unchanged_beliefs")
+@with_appcontext
+@click.option(
+    "--sensor-id",
+    type=int,
+    help="Delete unchanged (time series) data for a single sensor only. Follow up with the sensor's ID. ",
+)
+def delete_unchanged_beliefs(
+    sensor_id: Optional[int] = None,
+):
+    """Delete unchanged beliefs (i.e. updated beliefs with a later belief time, but with the same event value)."""
+    q = db.session.query(TimedBelief)
+    if sensor_id:
+        sensor = Sensor.query.filter(Sensor.id == sensor_id).one_or_none()
+        if sensor is None:
+            print(f"Failed to delete any beliefs: no sensor found with id {sensor_id}.")
+            return
+        q = q.filter(TimedBelief.sensor_id == sensor.id)
+    num_beliefs_before = len(q.all())
+    q_unchanged_beliefs = query_unchanged_beliefs(db.session, TimedBelief, q)
+    beliefs_up_for_deletion = q_unchanged_beliefs.all()
```

Comment:
We might get to the prompt faster if we first do a `.count()` here instead of `.all()`